Clarify Amazon Lambda invocation and sensing (#34653)
mistercrunch/Airflow: docs/apache-airflow-providers-amazon/operators/lambda.rst

To invoke an AWS lambda function you can use
:class:`~airflow.providers.amazon.aws.operators.lambda_function.LambdaInvokeFunctionOperator`.

.. note::
    According to `Lambda.Client.invoke <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda/client/invoke.html>`_ documentation
    for synchronous invocation (``invocation_type="RequestResponse"``) with a long timeout, your client might
    disconnect during synchronous invocation while it waits for a response.

    If you encounter ``ReadTimeoutError`` exceptions, like that

    .. code-block:: text

      urllib3.exceptions.ReadTimeoutError: AWSHTTPSConnectionPool(host='lambda.us-east-1.amazonaws.com', port=443): Read timed out. (read timeout=60)

>     Than you need configure :ref:`howto/connection:aws` to allow ``botocore`` / ``boto3`` use long connections with

Reviewer: ```suggestion
    Then you need to configure :ref:`howto/connection:aws` to allow ``botocore`` / ``boto3`` to use long connections with
```

>     If you encounter ``ReadTimeoutError`` exceptions, like that
> 
>     .. code-block:: text
> 
>       urllib3.exceptions.ReadTimeoutError: AWSHTTPSConnectionPool(host='lambda.us-east-1.amazonaws.com', port=443): Read timed out. (read timeout=60)
> 
>     Then you need to configure :ref:`howto/connection:aws` to allow ``botocore`` / ``boto3`` to use long connections with
>     timeout or keep-alive settings, in order to do this you need to add in Connection Extra
> 

Reviewer 2: Non-blocking nitpick:

```suggestion
    If this happens you will see a ``ReadTimeoutError`` exception similar to this:

    .. code-block:: text

      urllib3.exceptions.ReadTimeoutError: AWSHTTPSConnectionPool(host='lambda.us-east-1.amazonaws.com', port=443): Read timed out. (read timeout=60)

    If you encounter this issue, configure :ref:`howto/connection:aws` to allow ``botocore`` / ``boto3`` to use long connections with
    timeout or keep-alive settings in the Connection Extras field

```

    .. code-block:: json

        {
          "config_kwargs": {
            "connect_timeout": 900,
            "read_timeout": 900,
            "tcp_keepalive": true
          }
        }

>     In additional you might need to configure your firewall, proxy,

Reviewer: ```suggestion
    In addition, you might need to configure your firewall, proxy,
```

    or operating system to allow for long connections with timeout or keep-alive settings.

.. note::

>     You cannot verify the asynchronous invocation (``invocation_type="Event"``) of an AWS Lambda function.

Reviewer: ```suggestion
    You cannot describe the asynchronous invocation (``invocation_type="Event"``) of an AWS Lambda function.
```

.. _howto/sensor:LambdaFunctionStateSensor:

> Wait on an AWS Lambda function deployment state

Reviewer 2: Thanks for fixing this.   Amazon/AWS Service names are so confusing :+1:

Author: "There are only two hard things in Computer Science: cache invalidation and naming things"
